azure/services/bastionhosts/bastionhosts.go

```diff
@@ -17,88 +17,37 @@ limitations under the License.
 package bastionhosts
 
 import (
-	"context"
-
-	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
+	asonetworkv1 "github.com/Azure/azure-service-operator/v2/api/network/v1api20220701"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
-	"sigs.k8s.io/cluster-api-provider-azure/azure/services/async"
-	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
-	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/aso"
 )
 
 const serviceName = "bastionhosts"
 
 // BastionScope defines the scope interface for a bastion host service.
 type BastionScope interface {
 	azure.ClusterScoper
-	azure.AsyncStatusUpdater
-	AzureBastionSpec() azure.ResourceSpecGetter
+	aso.Scope
+	AzureBastionSpec() azure.ASOResourceSpecGetter[*asonetworkv1.BastionHost]
 }
 
 // Service provides operations on Azure resources.
 type Service struct {
 	Scope BastionScope
-	async.Reconciler
+	*aso.Service[*asonetworkv1.BastionHost, BastionScope]
 }
 
 // New creates a new service.
-func New(scope BastionScope) (*Service, error) {
-	client, err := newClient(scope)
-	if err != nil {
-		return nil, err
+func New(scope BastionScope) *Service {
+	svc := aso.NewService[*asonetworkv1.BastionHost, BastionScope](serviceName, scope)
+	spec := scope.AzureBastionSpec()
+	if spec != nil {
+		svc.Specs = []azure.ASOResourceSpecGetter[*asonetworkv1.BastionHost]{spec}
```

Comment:
```suggestion
	svc.Specs := scope.AzureBastionSpec()
```

Since svc.Specs is not set to empty `[]azure.ASOResourceSpecGetter[*asonetworkv1.BastionHost]`, we might as well set `svc.Specs` to the output of `scope.AzureBastionSpec()`

Reply:
I kept it this way because we would still need to wrap scope.AzureBastionSpec() in a []azure.ASOResourceSpecGetter, and creating a list with a nil value causes some issues.

Reply:
I apologize, I overlooked the fact that `scope.AzureBastionSpec()` returns a _single_ spec instead of a list of specs. This makes sense.